main.tf

```diff
@@ -18,19 +18,22 @@ locals {
   repo = var.repository != "" ? var.repository : "gcr.io/${var.project_id}"
 }
 
-provider "ko" {
-  repo = local.repo
-}
+provider "ko" {}
 
 // Create a service account for the prober to run as.
 resource "google_service_account" "prober" {
+  // Only create this if the user isn't specifying their
+  // own service account to run as.
+  count = var.service-account != "" ? 0 : 1
+
   project = var.project_id
-  # Service accounts can be 30 characters long, so truncate var.name to 23 chars.
-  account_id = "${substr(var.name, 0, 23)}-prober"
+  # Service accounts can be 30 characters long, so truncate var.name to 26 chars.
+  account_id = "${substr(var.name, 0, 26)}-prb"
```

Comment:
The `count` change will trigger a delete/create, so this change is intentional in order to avoid a name collision with the tombstoned service account, since I don't believe TF is smart enough to simple bring back the tombstoned GSA.

Reply:
Hmm, I am tempted to just mail this required moving forward since the most simple case of feeding a GSA `.email` in here leads to:
```
Error: Invalid count argument

  on .terraform/.../main.tf line 25, in resource "google_service_account" "prober":
  25:   count = var.service-account != "" ? 0 : 1

The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first apply only the resources that the
count depends on.
```